caffe2/operators/hip/conv_transpose_op_miopen.hip

```diff
@@ -298,10 +298,15 @@ bool MIOPENConvTransposeGradientOp<T>::RunOnDevice() {
   auto& Weight = Input(FILTER);
   auto& dY = Input(OUTPUT_GRAD);
   auto* dW = Output(FILTER_GRAD);
-  auto* dX = Output(no_bias_ ? BIAS_OR_INPUT_GRAD : INPUT_GRAD);
-  dX->ResizeLike(X);
   dW->ResizeLike(Weight);
 
+  bool doBwdDataComputation = (OutputSize() == 3 || (no_bias_ && (OutputSize() == 2)));
```

Comment:
This makes my eyes bleed lol

Reply:
I'd feel better if you just separately cased on the `no_bias` and `!no_bias` cases. Right now it feels like you're relying on some invariant where `no_bias_ && OutputSize() == 3` is impossible but it'd be easier to understand if this was spelled out explicitly

Reply:
...and I see below that this is preexisting code lol